metaserver/src/main/scala/org/langmeta/lsp/Commands.scala

```diff
@@ -0,0 +1,207 @@
+package org.langmeta.lsp
+
+import io.circe.Json
+import io.circe.generic.JsonCodec
+
+/**
+ * Parameters and types used in the `initialize` message.
+ */
+@JsonCodec case class InitializeParams(
+    /**
+     * The process Id of the parent process that started
+     * the server.
+     */
+    processId: Long,
+    /**
+     * The rootPath of the workspace. Is null
+     * if no folder is open.
+     */
+    rootPath: String,
+    /**
+     * The capabilities provided by the client (editor)
+     */
+    capabilities: ClientCapabilities
+)
+
+@JsonCodec case class ClientCapabilities()
+
+@JsonCodec case class ServerCapabilities(
+    /**
+     * Defines how text documents are synced.
+     */
+    textDocumentSync: TextDocumentSyncKind = TextDocumentSyncKind.Full,
+    /**
+     * The server provides hover support.
+     */
+    hoverProvider: Boolean = false,
+    /**
+     * The server provides completion support.
+     */
+    completionProvider: Option[CompletionOptions] = None,
+    /**
+     * The server provides signature help support.
+     */
+    signatureHelpProvider: Option[SignatureHelpOptions] = None,
+    /**
+     * The server provides goto definition support.
+     */
+    definitionProvider: Boolean = false,
+    /**
+     * The server provides find references support.
+     */
+    referencesProvider: Boolean = false,
+    /**
+     * The server provides document highlight support.
+     */
+    documentHighlightProvider: Boolean = false,
+    /**
+     * The server provides document symbol support.
+     */
+    documentSymbolProvider: Boolean = false,
+    /**
+     * The server provides workspace symbol support.
+     */
+    workspaceSymbolProvider: Boolean = false,
+    /**
+     * The server provides code actions.
+     */
+    codeActionProvider: Boolean = false,
+    /**
+     * The server provides code lens.
+     */
+    codeLensProvider: Option[CodeLensOptions] = None,
+    /**
+     * The server provides document formatting.
+     */
+    documentFormattingProvider: Boolean = false,
+    /**
+     * The server provides document range formatting.
+     */
+    documentRangeFormattingProvider: Boolean = false,
+    /**
+     * The server provides document formatting on typing.
+     */
+    documentOnTypeFormattingProvider: Option[DocumentOnTypeFormattingOptions] =
+      None,
+    /**
+     * The server provides rename support.
+     */
+    renameProvider: Boolean = false,
+    /**
+     * The server provides execute command support.
+     */
+    executeCommandProvider: ExecuteCommandOptions = ExecuteCommandOptions(Nil)
```

Comment:
Are `Command` in the filename and `Command` in LSP parlance (as referred to here, for example) the same thing?

Reply:
No. Command here is on another level of semantics, which is confusing but also consistent with the lsp spec.

Reply:
Is `Command` as in the filename every called "command" in the spec? I can only see a reference to one kind of commands in that document.

Reply:
I looked at https://github.com/Microsoft/vscode-languageserver-node and there seems to be a "protocol" which corresponds to "Commands.scala" and "types" which coresponds to our "types.scala".

I'll rename the file.

Sidenote, I was able to generate a json schema file from the typescript api, it would be cool to auto-generate scala sources with that.